OpenRA.Game/Selection.cs

```diff
@@ -57,9 +57,13 @@ public void Combine(World world, IEnumerable<Actor> newSelection, bool isCombine
 				}
 			}
 
-			var voicedActor = actors.FirstOrDefault(a => a.Owner == world.LocalPlayer && a.IsInWorld && a.HasTrait<IVoiced>());
-			if (voicedActor != null)
-				voicedActor.PlayVoice("Select");
+			var voicedActors = actors.Where(a => a.Owner == world.LocalPlayer && a.IsInWorld && a.HasTrait<IVoiced>());
+			var selectableActors = voicedActors.ToDictionary(a => a, v => v.TraitOrDefault<Selectable>());
+			foreach (var selectable in selectableActors.Where(v => v.Value != null))
+			{
+				selectable.Key.PlayVoice(selectable.Value.Info.Voice);
+				break;
+			}
```

Comment:
This can be simplified to 
```csharp
// Play the selection voice from one of the selected actors
// TODO: This probably should only be considering the newly selected actors
// TODO: Ship this into an INotifySelection trait to remove the engine dependency on Selectable
foreach (var a in actors)
{
	if (a.Owner != world.LocalPlayer || !a.IsInWorld)
		continue;

	var selectable = a.Info.Traits.GetOrDefault<SelectableInfo>();
	if (selectable == null || !a.HasVoice(selectable.Voice))
		continue;

	a.PlayVoice(selectable.Voice); 
	break;
}
```